### ISSUE TITLE
selected shapes not cleared

### ISSUE THREAD

**Issue by meguiraun** [mxcube/mxcube3#740]
hi, I noticed some weird issues with adding tasks and dig into it. It seems that the `selected_shapes` list on the interface is not cleared for each sample. So, if you do not deselect a point, when you add tasks to the next sample ... boom .. weirdness hits your face. 

What I saw (and our users suffered):
- Adding a task adds more than one, although they are not displayed in the interface. You also see un_number increasing by two instead of one
- Cannot add helical tasks

I think I fixed it here, I will submit PR as soon as I have it properly tested.

**Comment by marcus-oscarsson** [comment]
Hi Mikel,

First of all which version / commit are you using. We have been using the version on master all week and we have not seen this issue. We have been mounting and un-mounting samples and performing helical scans. 

It has maybe something to do with autoloop centring, its something that I imagine that we have that you perhaps don't have at the moment ?. Perhaps the autoloop centring fails and the points are not removed as a consequence ...

"That weirdness hits your face" seems pretty bad especially if its with a "boom" however it does unfortunately not tell us what to look for, but Ill in anyhow be looking for those symptoms to see if its something that we have here. For the moment I would say that its related to the differences between our setups because we, as far as I know, are not seeing this.

Slightly of topic, I have a few fixes to apply to master and that will become 3.0.0-beta2.

Marcus

**Comment by meguiraun** [comment]
I was feeling poetic.

What I saw is:
- Adding a task adds more than one, although they are not displayed in the interface. You also see run_number increasing by two instead of one, or even by higher number (once I saw +3)
- Cannot add helical tasks

We also have been doing collections on our own but we are not good enough to replicate all of this behaviour properly, the users tend to do many centrings, they collect on many points in the same sample, etc.

Although if you left a point selected when you add a new sample you will see that selected_shapes is not empty in the client, which I believe is the cause of the problems. I saw this a while ago but could not figure our the reason. But, perhaps it does not happen if you are using a sample changer?

It could be worse for us due to the differences on how we operate (we do a more manual way), but definitively there is an issue in master (we are not far behind). I will also commit asap...
